### PR TITLE
MAT-5992: Hide Attribute Tab when Not Needed

### PR DESCRIPTION
**Pull request by ethankaplan**
## MADiE PR

Jira Ticket: [MAT-5992](https://jira.cms.gov/browse/MAT-5992)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included
* [ ] No extemporaneous files are included (i.e Complied files or testing results)
* [ ] This PR is into the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/DataElementsCard.tsx [hunk 1]
```diff
@@ -199,7 +199,17 @@ const DataElementsCard = (props: {
     }
   }, [localSelectedDataElement, codeSystemMap, dataElements]);
   // centralize state one level up so we can conditionally render our child component
-
+  useEffect(() => {
+    setAttributesPresent(false);
+    if (selectedDataElement && selectedDataElement.schema?.eachPath) {
+      selectedDataElement.schema.eachPath((path) => {
```
Comment (adongare) [inline src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/DataElementsCard.tsx:205]: how about using https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some?

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/DataElementsCard.tsx [hunk 2]
```diff
@@ -293,6 +302,7 @@ const DataElementsCard = (props: {
           attributeChipList={displayAttributes}
           selectedDataElement={localSelectedDataElement}
           onDeleteAttributeChip={onDeleteAttributeChip}
+          setAttributesPresent={setAttributesPresent}
```
Comment (nmorasb) [inline src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/DataElementsCard.tsx]: what is this prop needed for within the AttributesSection? It doesn't appear to call this function

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/attributes/AttributeSection.tsx [hunk 3]
```diff
@@ -20,11 +20,13 @@ interface AttributeSectionProps {
   onAddClicked?: (attribute, type, attributeValue) => void;
   attributeChipList?: Array<Chip>;
   onDeleteAttributeChip?: (deletedChip) => void;
+  setAttributesPresent: (bool) => void;
 }
 
 const AttributeSection = ({
   attributeChipList = [],
   selectedDataElement,
+  setAttributesPresent,
```
Comment (nmorasb) [inline src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/attributes/AttributeSection.tsx]: It doesn't seem like this callback function is ever used. Is it needed?
And if not, let's remove it here as well as from the props interface a few lines above.

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/DataElementsCard.tsx (hunk 1 above)
Comment (adongare) [reply src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/DataElementsCard.tsx:205]: `eachPath` is from models. I did not realize that. Please ignore my comment. my bad.

**Review by nmorasb — APPROVED** [review]
